pitest-maven-verification/src/test/java/org/pitest/PitMojoIT.java

```diff
@@ -398,32 +323,73 @@ private File buildFile(File base, String... pathParts) {
+   * Sets up a test of the reporting mojo simulating multiple runs of mvn install (as in the case where timestampedReports is set to true).  
+   * First calls {@link #prepareSiteTest(String)} then walks all directories starting at target/pit-reports setting their last modified time 
+   * to 0 (epoch time).  Finally, the directory specified in the {@code latestDir} parameter has its last modified time set to {@link System#currentTimeMillis()}.
+   * 
+   * @param testPath {@link String} see {@link #prepareSiteTest(String)}
+   * @param latestDir {@link String} containing the subdirectory of target/pit-reports that should be set as the latest, 
+   *                  pass an empty string to indicate the target/pit-reports directory should be the latest   
+   * 
+   * @return {@link File} representing the temporary folder that was set up for this execution of the test
+   * @throws Exception
    */
-  private void waitUntilNextMinute(String startDateTime) throws Exception {
-    LOGGER.info("wait for next minute");
-
-    //this code ensures that will not happen
-    int loopCount = 0;
-    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmm");
-    while(dateFormat.format(new Date()).equals(startDateTime)){
-    	if(loopCount > 65){
-    		throw new PitError("integration test is stuck in an infinite loop");
-    	}
-
-    	Thread.sleep(1000);
-    	loopCount++;
+  private File prepareSiteTest(String testPath, String latestDir) throws Exception {
+    File testDir = prepareSiteTest(testPath);
+    File testTargetDir = this.buildFile(testDir, "target", "pit-reports"); //location where the target directory would be if a mvn clean install was executed
+    DirectoriesOnlyWalker walker = new DirectoriesOnlyWalker();
+    
+    for(File f : walker.locateDirectories(testTargetDir)) {
+        f.setLastModified(0L);
     }
+    
+    buildFile(testTargetDir, latestDir).setLastModified(System.currentTimeMillis());
+    
+    return testDir;
   }
+  
+  /**
+   * Sets up a test of the reporting mojo by simulating what a mvn clean install would do.  After this function is executed, 
+   * the maven site can be generated by executing the site goal.  
+   * 
+   * The provided {@code testPath} must have this folder structure set up underneath it:  src/test/resources/pit-reports  
+   * The contents of this directory will be moved to the target directory simulating a mvn clean install.   
+   * 
+   * @param testPath {@link String} location of the test to set up, this path is relative to <code>${basedir}/src/test/resources</code>
+   * 
+   * @return {@link File} representing the temporary folder that was set up for this execution of the test
+   * @throws Exception
+   */
+  private File prepareSiteTest(String testPath) throws Exception {
+      File testDir = prepare(testPath); //temporary location where the test will execute
```

Comment:
can we rename `testDir` to `temporaryTestLocation` to make the comment needless?